src/DateTime/DateAdjusted.ts

```diff
@@ -0,0 +1,18 @@
+import type { Moment } from "moment";
+import { getSettings } from "../Config/Settings";
+
+/**
+ * Returns the current moment, adjusted for the "Next day start hour" setting.
+ *
+ * If the current time is before the "next day start hour" (e.g., 2 AM),
+ * this function will return a moment object for the end of the *previous* day.
+ * Otherwise, it returns the current, unaltered moment.
+ */
+export function momentAdjusted(): Moment {
```

Comment:
There really needs to be tests for this, and updated versions of the existing tests for features that have been modified in this PR - to demonstrate how the behaviour has changed.

For example, demonstration of how this feature interacts in time-zones other than GMT... (I haven't yet been able to get my head around that)